### DISCUSSION TITLE
Fails to update to the recent release

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#458] [General]
Upon running ep_setup.exe, the following message appears and the version remains old:
![изображение](https://user-images.githubusercontent.com/1271736/143090131-39b9c6a9-e7ac-4a52-844f-19488cd051d5.png)

**Comment by valinet** [comment]
Have you applied what's described in there? It's obvious why it happens in your setup: it cannot update it a million times without rebooting because you use it as shell extension and the old code is still in use in God knows how many applications which are running since you started the OS. Just apply the steps there, specifically the last one: reboot the computer. The third step also tells you what to do specifically in your case. Alternatively, kill all the apps that still lock the files containing the old code, if possible.

I mean, the message pretty clearly states why it happens, so I reiterate it, despite me being the one who wrote it there as well.

**Reply by Anixx** [reply]
Yes, rebooting helps. Relogin did not help though.

**Reply by Anixx** [reply]
I also noticed that after clicking "restart explorer", a message telling about buffer overflow often appears. Not a big issue, but just let you know.

**Reply by valinet** [reply]
> Yes, rebooting helps. Relogin did not help though.

Does it say anything about relogin in the message? Why not apply the steps described there in the first place, before claiming that it fails...? You know they were written specifically for your use case. Failing to apply them beforehand disappoints me yet again. Also, understand how a shell extension works, please, as it's important to be aware how programs running on your computer work. It's registered in HKLM, so it gets injected on all processes usually, even those not running on your current user account. That's why reboot is the only easy to apply solution when needing to update a second time and using it as a shell extension.

> I also noticed that after clicking "restart explorer", a message telling about buffer overflow often appears. Not a big issue, but just let you know.

A screenshot, please? EP has those pop-ups silenced, like, it just crashes if something that bad happens, so it shouldn't be it, maybe something else. Anyway, show me how it looks.

**Reply by Anixx** [reply]
Since that time I still had not reproduce this but the message box is the same as if you try to run a XAML app .exe in Explorer:

![изображение](https://user-images.githubusercontent.com/1271736/143413396-b5ae1e7c-3a79-4657-97ff-9c6ebd5fed1b.png)

P.S. I wonder, why these messages and Win7-style flyouts of system tray have Win11-style window borders? Where they take them from?

**Reply by valinet** [reply]
What does the message say? What is the problem…? I don’t understand. Try to isolate things, you have so many mods that it’s hard to tell which is causing what…
